WooCommerce/Classes/POS/ViewModels/ItemSelectorViewModel.swift

```diff
@@ -4,14 +4,30 @@ import protocol Yosemite.POSItem
 import protocol Yosemite.POSItemProvider
 
 final class ItemSelectorViewModel: ObservableObject {
-    let selectedItemPublisher: AnyPublisher<POSItem, Never>
+    enum ItemSelectorState {
+        // TODO:
+        // Differentiate between loading on entering POS mode and reloading, as the
+        // screens will be different:
+        // https://github.com/woocommerce/woocommerce-ios/issues/13286
+        case loading
+        // TODO:
+        // Differentiate between loaded with products vs with no products vs with no eligible products
+        // https://github.com/woocommerce/woocommerce-ios/issues/12815
+        // https://github.com/woocommerce/woocommerce-ios/issues/12816
```

Comment:
Depending on interpretation it (` Product list not empty, but no simple ones`) could also be considered an `error` state.

ViewModel controls the business logic and filters out ineligible products, if there are no eligible products, we could have `error = .noSimpleProducts` which then the view decides how to display. 

`loaded` state can then be left for an array of `[Items]`. If a provided array is empty, the view shows a default empty state.

Reply:
Yes, that's something that came to mind when setting the states, and it's something we may want to discuss further in p2 with Android as well.

I personally would consider an error only if fails to perform the intended task when we expect a success (eg: network error, fetch error, ... ). In the case of "no products" or "non eligible products" I would consider it's a expected state of the system since fetch was successful, however it just didn't pass the policy we've set for products to be loaded and something else needs to be done before retrying 🤔 

That said, I can see reasons to use both, so no strong preference either.

Reply:
Yes, we can discuss it later.

> I personally would consider an error only if fails to perform the intended task when we expect a success

True, my only advice would be to represent this case in an explicit way.

We could have different types of Empty states: `state.empty(.noItems)`, `state.empty(.noEligibleItems)` which the `View` could decide how to represent. 

If we don't want to introduce `empty` state, then `error` state could be appropriate:

- From `View Model` perspective, successfully fetching and filtering out non-eligible items is indeed an expected state. 
- From `View` perspective, it's agnostic whether we cannot provide items due to network failure or due to no eligible products. It only cares that in case of `.loaded(items)` state -> present a list, and in case of `.error` state -> show a message with CTA.